mkosi

```diff
@@ -2313,14 +2313,27 @@ def detect_distribution():
 
     id = None
     version_id = None
+    codename = None
 
     for ln in f:
         if ln.startswith("ID="):
             id = ln[3:].strip()
         if ln.startswith("VERSION_ID="):
             version_id = ln[11:].strip()
+        if ln.startswith("VERSION="):
+            # extract Debain release codename
+            version_str = ln[8:].strip()
+            open_paren_pos = version_str.index("(")
+            close_paren_pos = version_str.index(")")
```

Comment:
there must be a more pythonic way to do something like this... in particular as it might do weird stuff if ) appears before (...

@keszybz can you suggest a pythonic way to solve this?